### PR TITLE
Add Recommended Tools article

### PR DESCRIPTION
**Pull request by aldeed**
Impact: **minor**  
Type: **docs**

## Changes
- Add "Recommended Tools" article
- Update/add some content in the Jest integration tests article

### CODE REVIEW

File: public-docs/recommended-tools.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+- [Promise Snippets](https://marketplace.visualstudio.com/items?itemName=progre.promise-snippets)
+- [ReactJS Code Snippets](https://marketplace.visualstudio.com/items?itemName=xabikos.ReactSnippets)
+- [Remark](https://marketplace.visualstudio.com/items?itemName=mrmlnc.vscode-remark)
+- [Simple React Snippets](https://marketplace.visualstudio.com/items?itemName=burkeholland.simple-react-snippets)
+- [Commitizen Support](https://marketplace.visualstudio.com/items?itemName=KnisterPeter.vscode-commitizen)
+- [styled-components](https://marketplace.visualstudio.com/items?itemName=jpoissonnier.vscode-styled-components)
+
+### Emacs
+
+Coming soon!
```
Comment (aldeed) [inline public-docs/recommended-tools.md]: @nnnnat Can you add a commit listing your emacs config / extensions here?

File: public-docs/recommended-tools.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+### Vim
+
+Coming soon!
```
Comment (aldeed) [inline public-docs/recommended-tools.md]: @ticean Can you add a commit listing your emacs config / extensions here?

**Comment by zenweasel** [comment]
You can just add a section for Webstorm that says "No additional plugins/configuration needed. Works out of the box" ;)

File: public-docs/recommended-tools.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+---
+id: recommended-tools
+title: Recommended Tools
+---
+
+If you are interested in being able to quickly debug and develop Reaction, consider installing these useful tools to help you.
+
+## General
+
+- You should use [nvm](https://github.com/creationix/nvm) to install and manage NodeJS.
```
Comment (spencern) [inline public-docs/recommended-tools.md]: Good to know. I didn't realize nvm was still the recommended way to install node.

**Pull request merged**